### PR TITLE
feat(inputs.win_wmi): Add support for remote queries

### LINKED ISSUE #14942: Remote host WMI collection capabilty in the win_wmi plugin
### Use Case

The abilty for the WMI plugin https://github.com/influxdata/telegraf/blob/9a9bb16c08adab133f2d995d6e26bdad130999ef/plugins/inputs/win_wmi/win_wmi.go#L1 to collect data from a **_remote_** host.

### Expected behavior

The abilty to specify a remote host(s) to gather WMI data from

### Actual behavior

Only local host is supported so agent must run on host. No agentless capbilty. 

### Additional info

WMI is generally a datastore accessed rmeotly by numerous native windows tools. Seems logical to add this feature.

### PR DESCRIPTION
**Pull request by srebhan**
## Summary

This PR adds support to query remote hosts via WMI if a host (and credentials) are specified.

## Checklist

- [x] No AI generated code was used in this PR

## Related issues

resolves #14942

### CODE REVIEW

File: plugins/inputs/win_wmi/README.md [hunk 1]
```diff
@@ -19,12 +19,27 @@ See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
...
+
+[SECRETSTORE]: ../../../docs/CONFIGURATION.md#secret-store-secrets
+
 ## Configuration
 
 ```toml @sample.conf
 # Input plugin to query Windows Management Instrumentation
 # This plugin ONLY supports Windows
 [[inputs.win_wmi]]
+  ## Hostname or IP for remote connections, by default the local machine is queried
```
Comment (powersj) [inline plugins/inputs/win_wmi/README.md:37]: I kind of wonder if this should be more explicit and say "for local machines leave blank"?

**Review by powersj — APPROVED** [review]
One comment on the config, otherwise +1

**Review by DStrand1 — APPROVED** [review]

**Review by DStrand1 — APPROVED** [review]

**Comment by powersj** [comment]
@srebhan reassigning this to you, didn't realize Dane had approved it. You can either update the readme with the one comment or land as-is. Thanks!

File: plugins/inputs/win_wmi/README.md (hunk 1 above)
Comment (shockogit) [reply plugins/inputs/win_wmi/README.md:37]: It's documented on the comments in the sample .conf file isn't it so i would think OK as is.

**Pull request merged**